app/workers/new_page_worker.rb

```diff
@@ -0,0 +1,44 @@
+class NewPageWorker
+  include Sidekiq::Worker
+  attr_reader :page, :doc_html
+
+  def perform(page_id)
+    @page = Page.find(page_id)
+    @doc_html = download_content
```

Comment:
move this 
@doc_html = download_content

to `process` after page.update_attributes(status: Page::IN_PROGRESS_STATUS)

because you already started some processing

Reply:
✓